packages/technical-features/messaging/agnostic/src/features/actual/computed-channel/computed-channel.injectable.ts

```diff
@@ -0,0 +1,209 @@
+
+export type ComputedChannelFactory = <T>(
+  channel: MessageChannel<T>,
+  pendingValue: T
+) => IComputedValue<T>;
+
+export const computedChannelInjectionToken =
+  getInjectionToken<ComputedChannelFactory>({
+    id: "computed-channel-injection-token",
+  });
+
+export type ChannelObserver<T extends Jsonifiable> = {
+  channel: MessageChannel<T>;
+  observer: IComputedValue<T>;
+};
+export type ComputedChannelAdminMessage = {
+  channelId: string;
+  status: "became-observed" | "became-unobserved";
+};
+
+export const computedChannelObserverInjectionToken = getInjectionToken<
+  ChannelObserver<Jsonifiable>
+>({
+  id: "computed-channel-observer",
+});
+
+const computedChannelInjectable = getInjectable({
+  id: "computed-channel",
+
+  instantiate: (di) => {
+    const sendMessageToChannel = di.inject(sendMessageToChannelInjectionToken);
+
+    return ((channel, pendingValue) => {
+      const observableValue = observable.box(pendingValue);
+
+      const computedValue = computed(() => {
+        const { trackingDerivation } = _getGlobalState();
+
+        const contextIsReactive = !!trackingDerivation;
+
+        if (!contextIsReactive) {
+          throw new Error(
+            `Tried to access value of computed channel "${channel.id}" outside of reactive context. This is not possible, as the value is acquired asynchronously sometime *after* being observed. Not respecting that, the value could be stale.`
+          );
+        }
+
+        return observableValue.get();
+      });
+
+      const valueReceiverInjectable = getMessageChannelListenerInjectable({
+        id: `computed-channel-value-receiver-for-${channel.id}`,
+        channel,
+
+        getHandler: () => (message) => {
+          runInAction(() => {
+            observableValue.set(message);
+          });
+        },
+      });
+
+      runInAction(() => {
+        di.register(valueReceiverInjectable);
+      });
+
+      onBecomeObserved(computedValue, () => {
+        runInAction(() => {
+          observableValue.set(pendingValue);
+        });
+
+        sendMessageToChannel(computedChannelAdministrationChannel, {
+          channelId: channel.id,
+          status: "became-observed",
+        });
+      });
+
+      onBecomeUnobserved(computedValue, () => {
+        runInAction(() => {
+          observableValue.set(pendingValue);
+        });
+
+        sendMessageToChannel(computedChannelAdministrationChannel, {
+          channelId: channel.id,
+          status: "became-unobserved",
+        });
+      });
+
+      return computedValue;
+    }) as ComputedChannelFactory;
+  },
+
+  injectionToken: computedChannelInjectionToken,
+});
+
+export const duplicateChannelObserverGuardInjectable = getInjectable({
+  id: "duplicate-channel-observer-guard",
+
+  instantiate: (di) => {
+    const computedInjectMany = di.inject(computedInjectManyInjectable);
+
+    return {
+      run: () => {
+        reaction(
```

Comment:
I don't think it is, since here we know that it is only ever called once in the lifecycle of the application. Disposing would happen when application is quit. Would add it though, just to be sure but at the moment we don't have anything related to application quit/stop extracted yet.